llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp

```diff
@@ -295,6 +298,23 @@ void LLVM_Hs_disposeLambdaResolver(LLVMLambdaResolverRef resolver) {
     delete resolver;
 }
 
+LLVMObjectHandle LLVM_Hs_LinkingLayer_addObject(LinkingLayer *linkLayer,
+                                                LLVMObjectFileRef  obj,
+                                                LLVMLambdaResolverRef resolver,
+                                                char **errorMessage) {
+    auto obj2 = reinterpret_cast<object::OwningBinary<object::ObjectFile> *>(obj);
+    auto obj3 = std::move(*obj2);
+    auto obj4 = std::make_shared<object::OwningBinary<object::ObjectFile>>(std::move(obj3));
```

Comment:
I know this terrible. Still working on it.

Reply:
`unwrap` will give you back a `OwningBinary<ObjectFile> *`, you can use the constructor of `shared_ptr`. Take a look at https://github.com/ckoparkar/llvm-hs/blob/92a6cf1b18a2bed65a77af20031e3dc5bc55e265/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp#L242 It should be quite similar (I haven’t actually tried this so if you run into problems let me know and I’ll take a closer look).

Reply:
I think I tried using unwrap at some point. And it spit out a a really long template error stacktrace. I probably didn't use it properly. Let me try again. Will post an update here.